src/manipulators/no_implicit_returns_manipulator.ts

```diff
@@ -0,0 +1,105 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+import _ from 'lodash';
+import {Diagnostic, ts, Node, SyntaxKind} from 'ts-morph';
+import {Manipulator} from './manipulator';
+import {DiagnosticCodes} from 'types';
+import {ErrorDetector} from 'error_detectors/error_detector';
+
+/**
+ * Manipulator that fixes for the noImplicitReturns compiler flag.
+ * @extends {Manipulator}
+ */
+export class NoImplicitReturnsManipulator extends Manipulator {
+  private nodeKinds: Set<SyntaxKind>;
+
+  constructor(errorDetector: ErrorDetector) {
+    super(
+      errorDetector,
+      new Set<number>([DiagnosticCodes.CodePathNoReturn])
+    );
+    this.nodeKinds = new Set<SyntaxKind>([
+      SyntaxKind.Identifier,
+      SyntaxKind.ReturnStatement,
+      SyntaxKind.ArrowFunction,
+    ]);
+  }
+
+  /**
+   * Manipulates AST of project to fix for the noImplicitReturns compiler flag given diagnostics.
+   * @param {Diagnostic<ts.Diagnostic>[]} diagnostics - List of diagnostics outputted by parser.
+   */
+  fixErrors(diagnostics: Diagnostic<ts.Diagnostic>[]): void {
+    // Retrieve AST nodes corresponding to diagnostics with relevant error codes.
+    const errorNodes = this.errorDetector.filterNodeDiagnosticsByKind(
```

Comment:
filterNodeDiagnosticsByKind -> sortAndFilterDiagnosticsByKind

so you don't need forEachRight any more. I assume you want right to left iteration for all manipulator.